source/contributing/implementation/datamanipulator.rst

```diff
@@ -0,0 +1,351 @@
+An updated list of DataManipulators to be implemented can be found at
+`SpongeCommon Issue #8 <https://github.com/SpongePowered/SpongeCommon/issues/8>`_.
+
+To fully implement a ``DataManipulator`` these steps must be followed:
+
+1. Implement the ``DataManipulator`` itself
+#. Implement the ``ImmutableDataManipulator``
+
+When these steps are complete, the following must also be done:
+
+3. Register the ``Key`` in the ``KeyRegistry``
+#. Implement the ``DataProcessor``
+#. Implement the ``DataBuilder``
+#. Implement the ``ValueProcessor`` for each value being represented by the ``DataManipulator``
+#. Register everything in ``SpongeGameRegistry``
+
+.. note::
+    Make sure you follow our :doc:`../guidelines`.
+
+1. Implement the DataManipulator
+================================
+
+The naming convention for ``DataManipulator`` implementations is the name of the interface prefixed with "Sponge".
+So to implement the ``HealthData`` interface, we create a class named ``SpongeHealthData`` in the appropriate package.
+For implementing the ``DataManipulator`` first have it extend an appropriate abstract class from the
+``org.spongepowered.common.data.manipulator.mutable.common`` package. The most generic there is ``AbstractData``
+but there are also abstractions that reduce boilerplate code even more for some special cases like
+``DataManipulator``\ s only containing a single value.
+
+.. code-block:: java
+
+    public class SpongeHealthData extends AbstractData<HealthData, ImmutableHealthData> implements HealthData {
+        [...]
+    }
+
+There are two type arguments to the AbstractData class. The first is the interface implemented by this class, the
+second is the interface implemented by the corresponding ``ImmutableDataManipulator``.
+
+The Constructor
+~~~~~~~~~~~~~~~
+
+In most cases while implementing an abstract Manipulator you want to have two constructors:
+
+* One without arguments (no-args) which calls the second
+* and one single-argument constructor which is actually used for the provided values
+
+The second constructor must
+
+* make a call to the ``AbstractData`` constructor, passing the class reference for the implemented interface.
+* call the ``registerGettersAndSetters()`` method
+
+.. code-block:: java
+
+    public SpongeHealthData() {
+        this(20D, 20D);
+    }
+
+    public SpongeHealthData(double currentHealth, double maxHealth) {
+        super(HealthData.class);
+        this.currentHealth = currentHealth;
+        this.maximumHealth = maxHealth;
+        this.registerGettersAndSetters();
+    }
+
+Accessors defined by the Interface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The interface we implement specifies some methods to access ``Value`` objects. For ``HealthData``, those are
+``health()`` and ``maxHealth()``. Every call to those should yield a new ``Value``.
+
+.. code-block:: java
+
+    public MutableBoundedValue<Double> health() {
+        return new SpongeBoundedValue<>(Keys.HEALTH, this.maximumHealth,
```

Comment:
You should likely put each argument being passed in a new line and have comments to describe the argument ordering.